src/utils/mutate.js

```diff
@@ -0,0 +1,249 @@
+import { chunk, cloneDeep, flatten, mapValues } from 'lodash';
+import { firestoreRef } from './query';
+import mark from './profiling';
+
+/**
+ * @param {object} firestore
+ * @param {string} collection
+ * @param {string} doc
+ * @returns Boolean
+ */
+const docRef = (firestore, collection, doc) =>
+  firestore.doc(`${collection}/${doc}`);
```

Comment:
Is there a reason to make this a separate function if we are just calling the existing method?

Reply:
What happens if/when the firestore API changes? It can be changed just here (because it wraps the library) or change it in multiple places. 

That’s the reason I tried to keep a call to Firestore calls to a minimum. It makes it easier to support when new versions come out. Also a backward compatible of multiple versions is easier because it’s just a single if/else here.